tests/unit/test-firefox/test.firefox.js

```diff
@@ -263,6 +294,113 @@ describe('firefox', () => {
 
   });
 
+  describe('isDefaultProfile', () => {
+
+    it('detects common Firefox default profiles specified by name',
+       async () => {
+         const isDefault = await firefox.isDefaultProfile('default');
+         assert.equal(isDefault, true);
+
+         const isDevEditionDefault = await firefox.isDefaultProfile(
+           'dev-edition-default'
+         );
+         assert.equal(isDevEditionDefault, true);
+       });
+
+    it('allows profile name if is not listed as default in profiles.ini',
+      async () => {
+        return withTempDir(async (tmpDir) => {
+          const profilesDirPath = tmpDir.path();
+          const FakeProfileFinder = createFakeProfileFinder(profilesDirPath);
```

Comment:
Instead of using a fake `ProfileFinder`, could you just pass in an explicit `baseProfileDir` value to `firefox.isDefaultProfile()`? That would bypass `ProfileFinder.locateUserDirectory()` but continue to use a real `ProfileFinder` which would provide better test coverage.